### PR TITLE
Store Picker: Add UI for new screen to edit store list

### PR DESCRIPTION
**Pull request by itsmeichigo**
Part of #14658 

## Description

This PR adds the UI for the Edit Store List view. The UI follows the mockup in #14658 but I updated the layout a bit to better match the iOS platform style.

The actual logic for saving selections will be handled in a future PR. 

## Steps to reproduce

- Enable the feature flag `hideSitesInStorePicker` and build the app.
- Log in to a test store and navigate to the menu tab.
- Select Edit on the top right.
- Confirm that a new screen is presented with the list of all stores and checkboxes to select stores to display except for the currently selected store. 
- Tap Save or Dismiss and confirm that the screen is dismissed.

## Testing information

Tested on simulator iPhone 16 Pro and iPad Mini iOS 18.1 in light and dark mode and confirmed that the layout looks good.

## Screenshots

iPhone | iPad
--- | ---
 | 

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/233">21.3</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/WooCommerceTests/Authentication/EditStoreListViewModelTests.swift [hunk 1]
```diff
@@ -0,0 +1,83 @@
+import Testing
+import struct Yosemite.Site
+@testable import WooCommerce
+
+struct EditStoreListViewModelTests {
+    // Given
+    private let site1 = Site.fake().copy(siteID: 123)
+    private let site2 = Site.fake().copy(siteID: 135)
+
+    @Test func hasChanges_returns_correct_values() {
```
Comment (selanthiraiyan) [inline WooCommerce/WooCommerceTests/Authentication/EditStoreListViewModelTests.swift:10]: I didn't know that we were already using Swift Testing in our repo. Nice.

**Review by selanthiraiyan — APPROVED** [review]
Nice work, Huong! Code LGTM and tests as expected. 🚀

**Pull request merged**